sonar-plugin-api/src/main/java/org/sonar/api/scan/filesystem/PathResolver.java

```diff
@@ -58,6 +48,10 @@ public File relativeFile(File dir, String path) {
     return result;
   }
 
+  /**
+   * @deprecated since 6.0 was used when component keys were relative to source dirs
+   */
+  @Deprecated
   @CheckForNull
   public RelativePath relativePath(Collection<File> dirs, File file) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)